### PR TITLE
mangareader: init at 2.0.4

### PR DESCRIPTION
**Pull request by zendo**
###### Description of changes

Qt manga reader for local files. 
https://github.com/g-fb/mangareader

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/mangareader/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, extra-cmake-modules
+, cmake
+, qt5
```
Comment (azahi) [inline pkgs/applications/graphics/mangareader/default.nix:6]:
```suggestion
, wrapQtAppsHook
```

File: pkgs/applications/graphics/mangareader/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    owner = "g-fb";
+    repo = pname;
+    rev = "${version}";
+    sha256 = "sha256-ow1EjT7WKEs9IKbRvh95ffMv8lVVgSsOWAx6iF8/D0w=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    extra-cmake-modules
+    qt5.wrapQtAppsHook
```
Comment (azahi) [inline pkgs/applications/graphics/mangareader/default.nix:30]:
```suggestion
    wrapQtAppsHook
```

File: pkgs/applications/graphics/mangareader/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mangareader";
+  version = "2.0.3";
+
+  src = fetchFromGitHub {
+    owner = "g-fb";
+    repo = pname;
+    rev = "${version}";
```
Comment (azahi) [inline pkgs/applications/graphics/mangareader/default.nix:23]:
```suggestion
    rev = version;
```

File: pkgs/applications/graphics/mangareader/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    kxmlgui
+    kconfig
+    karchive
+    kcoreaddons
+    kconfigwidgets
+  ];
+
+  meta = with lib; {
+    description = "Qt manga reader for local files";
+    homepage = "https://github.com/g-fb/mangareader";
```
Comment (azahi) [inline pkgs/applications/graphics/mangareader/default.nix:45]:
```suggestion
    homepage = "https://github.com/g-fb/mangareader";
    changelog = "https://github.com/g-fb/mangareader/releases/tag/${src.rev}"
```

File: pkgs/applications/graphics/mangareader/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  buildInputs = [
+    kio
+    ki18n
+    kxmlgui
+    kconfig
+    karchive
+    kcoreaddons
+    kconfigwidgets
+  ];
+
```
Comment (azahi) [inline pkgs/applications/graphics/mangareader/default.nix:42]:
```suggestion

  passthru.updateScript = nix-update-script {
    attrPath = pname;
  };
  
```

File: pkgs/applications/graphics/mangareader/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+, extra-cmake-modules
+, cmake
+, qt5
+, kio
+, ki18n
+, kxmlgui
+, kconfig
+, karchive
+, kcoreaddons
+, kconfigwidgets
```
Comment (azahi) [inline pkgs/applications/graphics/mangareader/default.nix:13]:
```suggestion
, kconfigwidgets
, nix-update-script
```

**Review by azahi — CHANGES_REQUESTED** [review]

**Comment by ajs124** [comment]
2.0.4 is already out

**Pull request merged**